src/SparqlLua.php

```diff
@@ -4,27 +4,22 @@
 
 namespace ProfessionalWiki\SPARQL;
 
+use MediaWiki\MediaWikiServices;
 use Scribunto_LuaLibraryBase;
 
 class SparqlLua extends Scribunto_LuaLibraryBase {
 
 	/**
-	 * @return mixed[]
+	 * @return array<mixed>
 	 */
 	public function register(): array {
+		$queryRunner = new SparqlLuaQueryRunner( MediaWikiServices::getInstance()->getHttpRequestFactory() );
 		return $this->getEngine()->registerInterface(
 			__DIR__ . '/Sparql.lua',
 			[
-				'runQuery' => fn( string $subjectId ): array => [ $this->runQuery( $subjectId ) ],
+				'runQuery' => fn ( string $subjectId ): array => [ $queryRunner->runQuery( $subjectId ) ],
```

Comment:
Fine as it is, but as a general note, I do think it is cleaner with the private function that you removed, so you separate the MediaWiki registration aspect from our own dependency graph building. To keep functions focused on one responsibility.

Reply:
Here is an example of us constructing one of our services with some MediaWiki config: https://github.com/ProfessionalWiki/ExternalContent/blob/master/src/EmbedExtensionFactory.php#L99-L103

In "modern" MediaWiki the idea is to access the config via `MediaWikiServices::getInstance()->getMainConfig()->get`, rather than via the $wgGlobal